stdlib/package.py

```diff
@@ -1,6 +1,198 @@
+        The default value is ``None``.
+    :param version: The version number. The version should follow `Semantic Versioning 2.0.0`_.
+        If ``None`` is given, the version number is taken from the current :py:class:`~stdlib.manifest.BuildManifestMetadata`.
+        The default value is ``None``.
+
+    .. _Semantic Versioning 2.0.0: https://semver.org/spec/v2.0.0.html
+    """
+    def __init__(
+        self,
+        name: str,
+        repository: str = None,
+        category: str = None,
+        version: str = None,
+    ):
+        build = stdlib.build.current_build()
+
+        import core.args
+        self.repository = repository if repository is not None else core.args.get_args().repository
+        self.category = category if category is not None else build.manifest.metadata.category
+        self.version = version if version is not None else build.semver
+        self.name = name
+
+    def full_name(self) -> str:
+        """Return a string representing the full name of the package, which is the combination of its repository, category and name.
+
+        :return: A string representing The full name of the package
+        """
+        return f'{self.repository}::{self.category}/{self.name}'
+
+    def __str__(self) -> str:
+        return f'{self.full_name()}#{self.version}'
 
 
 class Package():
-    pass
+    """A package, the output of a :py:class:`~stdlib.build.Build`.
+
+    :param id: The unique identifier of the package, including its name, category, repository and version.
+    :param description: A description of the package. This description should start with an uppercase letter and finish with a dot.
+        If ``None`` is given, the description is taken from the current :py:class:`~stdlib.manifest.BuildManifestMetadata`.
+        The default value is ``None``.
+    :param tags: A list of tags helping a user find the built packages easily. Each tag should be in ``snake-case``.
+        If ``None`` is given, the list of tags is taken from the current :py:class:`~stdlib.manifest.BuildManifestMetadata`.
+        The default value is ``None``.
+    :param maintainer: The email address of the maintainer of this build manifest.
+        If ``None`` is given, the maintainer's email address is taken from the current :py:class:`~stdlib.manifest.BuildManifestMetadata`.
+        The default value is ``None``.
+    :param licenses: The licenses of the built software.
+        If ``None`` is given, the licenses are taken from the current :py:class:`~stdlib.manifest.BuildManifestMetadata`.
+        The default value is ``None``.
+    :param upstream_url: An URL pointing to the home page of the software.
+        If ``None`` is given, the url is taken from the current :py:class:`~stdlib.manifest.BuildManifestMetadata`.
+        The default value is ``None``.
+    :param run_dependencies: A dictionnary of runtime dependencies. The key is a package full name, and the value is a version requirement.
```

Comment:
`a package's full name`
The same error appears also at the line 98 of this file